lib/Doctrine/ORM/Proxy/ProxyFactory.php

```diff
@@ -138,12 +138,13 @@ private function createInitializer(ClassMetadata $classMetadata, EntityPersister
                 $proxy->__setInitialized(true);
                 $proxy->__wakeup();
 
-                if (null === $entityPersister->loadById($classMetadata->getIdentifierValues($proxy), $proxy)) {
+                $id = $classMetadata->getIdentifierValues($proxy);
```

Comment:
Please compute the ID only when throwing the exception: these code-paths are performance-critical.